### PR TITLE
Refactor email otp authenticator to handle multi attribute login

### PR DESCRIPTION
**Pull request by sandushi**
## Purpose
- When multi attribute login is enabled, from the IDF hanlder the user is resolved using the multi attribute login service and set `isUserResolved : true` to the context.
- From these changes, the user is only resolved from the user store if the user is not resolved from the previous step when previous step is IDF. This is done to avoid the duplicate resolvings from the user store because user is already resolved in the IDF, when multi attribute login is enabled.

## Merge this after 
https://github.com/wso2/carbon-identity-framework/pull/4860

### CODE REVIEW

File: components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java [hunk 1]
```diff
@@ -240,6 +239,19 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                 response, request, context);
     }
 
+    private AuthenticatedUser resolveUserFromUserStore(AuthenticatedUser authenticatedUserFromContext)
+            throws AuthenticationFailedException {
+
+        User user;
+        user = getUser(authenticatedUserFromContext);
```
Comment (amanda-ariyaratne) [inline components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java:246]: Can directly assign where the user variable is declared.

File: components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java [hunk 2]
```diff
@@ -240,6 +239,19 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                 response, request, context);
     }
 
+    private AuthenticatedUser resolveUserFromUserStore(AuthenticatedUser authenticatedUserFromContext)
```
[the remark this answers is not in the document]
Comment (sandushi) [reply components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java]: Resolved by https://github.com/wso2-extensions/identity-local-auth-emailotp/pull/5/commits/fa9bd97219b16f6ab148004e38ac4c5a6b239c96

File: components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java (hunk 1 above)
Comment (sandushi) [reply components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java]: Resolved by https://github.com/wso2-extensions/identity-local-auth-emailotp/pull/5/commits/fa9bd97219b16f6ab148004e38ac4c5a6b239c96

File: components/org.wso2.carbon.identity.local.auth.emailotp/src/test/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticatorTest.java [hunk 3]
```diff
@@ -283,6 +279,12 @@ private void setStepConfigWithEmailOTPAuthenticator(AuthenticatorConfig authenti
         context.getSequenceConfig().setApplicationConfig(applicationConfig);
     }
 
+    private void mockMultiAttributeLoginService() {
+        when(FrameworkServiceDataHolder.getInstance()).thenReturn(frameworkServiceDataHolder);
+        when(frameworkServiceDataHolder.getMultiAttributeLoginService()).thenReturn(multiAttributeLoginService);
```
Comment (sahandilshan) [inline components/org.wso2.carbon.identity.local.auth.emailotp/src/test/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticatorTest.java:284]:
```suggestion
    private void mockMultiAttributeLoginService() {
    
        when(FrameworkServiceDataHolder.getInstance()).thenReturn(frameworkServiceDataHolder);
        when(frameworkServiceDataHolder.getMultiAttributeLoginService()).thenReturn(multiAttributeLoginService);
```

**Review by sahandilshan — APPROVED** [review]

File: components/org.wso2.carbon.identity.local.auth.emailotp/src/test/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticatorTest.java (hunk 3 above)
Comment (sandushi) [reply components/org.wso2.carbon.identity.local.auth.emailotp/src/test/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticatorTest.java]: Resolved by https://github.com/wso2-extensions/identity-local-auth-emailotp/pull/5/commits/7cea76fc926c8d8ea441d5b0486501d6ec4acb7f

File: components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java [hunk 4]
```diff
@@ -267,6 +266,18 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
...
     }
 
+    private AuthenticatedUser resolveUserFromUserStore(AuthenticatedUser authenticatedUser)
+            throws AuthenticationFailedException {
+
+        User user = getUser(authenticatedUser);
+        if (user != null) {
+            authenticatedUser = new AuthenticatedUser(user);
+        } else {
+            authenticatedUser = null;
```
[the remark this answers is not in the document]
Comment (sandushi) [reply components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java]: Resolved by https://github.com/wso2-extensions/identity-local-auth-emailotp/pull/5/commits/7cea76fc926c8d8ea441d5b0486501d6ec4acb7f

**Pull request merged**